src/Pact/Analyze/PrenexNormalize.hs

```diff
@@ -48,7 +42,9 @@ instance Float Time where
   float = floatTimeQuantifiers
 
 instance Float Object where
-  float p = case p of STANDARD_INSTANCES
+  float p = case p of
+    STANDARD_INSTANCES
+    PLiteralObject{} -> ([], p)
 
 instance Float KeySet where
   float p = case p of STANDARD_INSTANCES
```

Comment:
hmm is CPP really the only way to reuse the matches?

Reply:
I think CPP works quite well here